### PR TITLE
Add lint workflow

### PR DESCRIPTION
**Pull request by MrAlias**

### CODE REVIEW

**Comment by MrAlias** [comment]
Looks like the repository is not configured to allow actions to run.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      - main
+  pull_request:
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v5
+        with:
+          go-version: "~1.23.0"
```
Comment (damemi) [inline .github/workflows/lint.yml:14]: I'm not up on the ~ syntax, so does this run 1.23 or 1.22? I pinned the repo to 1.22: https://github.com/open-telemetry/govanityurls/blob/65456fc5e8ab7e7c375bab48c6c42b23396b7a28/go.mod#L3

**Review by damemi — APPROVED** [review]

File: .github/workflows/lint.yml (hunk 1 above)
Comment (MrAlias) [reply .github/workflows/lint.yml:14]: This does pin the Go version to 1.23+, but I'm not sure that this step is actually needed.

I'm going to instigate if this is even needed.

Comment (MrAlias) [reply .github/workflows/lint.yml]: [It looks unneeded](https://github.com/MrAlias/flow/pull/180). Removed.

**Comment by jsuereth** [comment]
your GHA should be running here, let me know if you still see issues.

**Pull request merged**